metals/src/main/scala/scala/meta/internal/metals/CallHierarchyProvider.scala

```diff
@@ -0,0 +1,653 @@
+package scala.meta.internal.metals
+
+import scala.meta.internal.metals.JsonParser._
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags.Semanticdbs
+import scala.meta.internal.semanticdb.TextDocuments
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.{semanticdb => s}
+
+import scala.meta.internal.parsing.Trees
+import scala.meta.internal.remotels.RemoteLanguageServer
+
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.lsp4j.CallHierarchyPrepareParams
+import org.eclipse.lsp4j.CallHierarchyIncomingCallsParams
+import org.eclipse.lsp4j.CallHierarchyOutgoingCallsParams
+import org.eclipse.lsp4j.CallHierarchyIncomingCall
+import org.eclipse.lsp4j.CallHierarchyOutgoingCall
+import org.eclipse.lsp4j.CallHierarchyItem
+import org.eclipse.{lsp4j => l}
+import scala.meta.Tree
+import scala.meta.Defn
+import scala.meta.Pat
+import scala.meta.Name
+import com.google.gson.JsonElement
+import scala.meta.Member
+import scala.meta.Term
+import scala.meta.pc.CancelToken
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.meta.Decl
+import scala.meta.Type
+import scala.meta.Init
+import scala.meta.Template
+
+final case class CallHierarchyProvider(
```

Comment:
It's a huge file with a lot of stuff going on. What do think about partitioning it a bit? 
- File itself can be moved to separate package ( `scala.meta.internal.metals.callHierarchy` maybe? Camel case isn't the best one for packages but in this case IMO it's better than `call.hierarchy` or similar)
- `CanBuildCallHierarchyItem` can be extracted to file and made package private to not pollute namespace
- there are a few tuples (e.g. return type of `findOutgoingCallsSynthetics` method) - can we have a case classes for them with some scaladoc? It'll make methods like `groupIncomingCalls` more readable

What do you think?

Reply:
Thanks for the advice, I finish reorganize the code, and adding some pieces of scaladoc.